### PR TITLE
Deque

### PR DESCRIPTION
**Pull request by scotist**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,15 @@
 # linked-list
 Implements a linked list with python
 
-This repo holds sample code for some classic data structures, implemented in Python, including singly- and doubly-linked lists and stacks. Singly-linked lists are convenient for keeping a group of data nodes in order; adding and removing them is simple and efficient. They are most useful when we only need to traverse our data in one direction. Doubly-linked lists make their nodes more accessible since they can be inserted or removed at either end of the list; but navigating the list is costlier.
+This repo holds sample code for some classic data structures, implemented in Python, including singly- and doubly-linked lists and stacks, queues, and deques. Singly-linked lists are convenient for keeping a group of data nodes in order; adding and removing them is simple and efficient. They are most useful when we only need to traverse our data in one direction. Doubly-linked lists make their nodes more accessible since they can be inserted or removed at either end of the list; but navigating the list is costlier. Stacks, queues, and deques are more refined implementations.
```
Comment (joelstanner) [inline README.md]: As your data structures repo gets larger, you might want to consider breaking this out into separate paragraphs and sections that describe each data structure. This paragraph will get pretty dense before long (plus employers like that).

File: src/deque.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class Deque(object):
+    """Make deque object."""
+    def __init__(self, val=None):
+        """Make new deque."""
+        self._container_q = Queue(val)
+
+    def pop(self):
+        """Pop item from end of deque."""
+        return self._container_q._container.shift()
```
Comment (joelstanner) [inline src/deque.py]: What happens when you try this pop method on a queue with one item in it? Write tests for edge cases. I'm getting an error on a 1 item list, but the `pop_left()` works.

File: src/test_deque.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# _*_ encoding: utf-8 _*_
+"""Test deque module in data structures."""
+import pytest
+
+APPEND = [([1, 2, 3, 4], 'addme', 'addme'),
+          (['Hey', 'there', 'Howdy'], 'addthis', 'addthis')]
+
+POP_PEEK = [([3, 6, 2, 7, 7, 9], 9),
+            ([4, 6, 2, 3], 3),
+            ([], None)]
```
Comment (joelstanner) [inline src/test_deque.py]: Add your 1-of test case here.
